File: services/education/internal/model/tuition_account_flow.go
package model

import "time"

const (
	TuitionAccountFlowSourceRegistration              = 1
	TuitionAccountFlowSourceTransferIn                = 2
	TuitionAccountFlowSourceCrossCampusTransferIn     = 3
	TuitionAccountFlowSourceCrossCampusAttendIn       = 4
	TuitionAccountFlowSourceConsumeReturn             = 5
	TuitionAccountFlowSourceRevokeGraduate            = 6
	TuitionAccountFlowSourceExpireRollback            = 7
	TuitionAccountFlowSourceRevokeRefundOrder         = 8
	TuitionAccountFlowSourceRevokeTransferOut         = 9
	TuitionAccountFlowSourceRevokeImportConsume       = 10
	TuitionAccountFlowSourceRevokeAutoConsume         = 11
	TuitionAccountFlowSourceConsume                   = 12
	TuitionAccountFlowSourceImportConsume             = 13
	TuitionAccountFlowSourceConsumeSupplement         = 14
	TuitionAccountFlowSourceAutoConsume               = 15
	TuitionAccountFlowSourceConsumeArrearsSettlement  = 16
	TuitionAccountFlowSourceTransferOut               = 17
	TuitionAccountFlowSourceCrossCampusTransferOut    = 18
	TuitionAccountFlowSourceCrossCampusAttendOut      = 19
	TuitionAccountFlowSourceGraduate                  = 20
	TuitionAccountFlowSourceExpireGraduate            = 21
	TuitionAccountFlowSourceRefund                    = 22
	TuitionAccountFlowSourceOrderVoid                 = 23
	TuitionAccountFlowSourceVoidCrossCampusTransferIn = 24
)

type TuitionAccountFlowRecordListQueryDTO struct {
	PageRequestModel PageRequestModel                   `json:"pageRequestModel"`
	QueryModel       TuitionAccountFlowRecordQueryModel `json:"queryModel"`
	SortModel        TuitionAccountFlowRecordSortModel  `json:"sortModel"`
}

type TuitionAccountFlowRecordQueryModel struct {
	ProductID   string `json:"productId"`
	StudentID   string `json:"studentId"`
	SourceTypes []int  `json:"sourceTypes"`
	StartTime   string `json:"startTime"`
	EndTime     string `json:"endTime"`
}

type TuitionAccountFlowRecordSortModel struct {
	OrderByCreatedTime int `json:"orderByCreatedTime"`
}

type TuitionAccountFlowRecordItem struct {
	TuitionAccountFlowID string     `json:"tutionAccountFlowId"`
	TuitionAccountID     string     `json:"tuitionAccountId"`
	StudentID            string     `json:"studentId"`
	StudentName          string     `json:"studentName"`
	StudentPhone         string     `json:"studentPhone"`
	StudentAvatar        string     `json:"studentAvatar"`
	ProductID            string     `json:"productId"`
	ProductName          string     `json:"productName"`
	LessonType           *int       `json:"lessonType,omitempty"`
	LessonChargingMode   *int       `json:"lessonChargingMode,omitempty"`
	SourceType           int        `json:"sourceType"`
	SourceID             string     `json:"sourceId"`
	TeachingRecordID     string     `json:"teachingRecordId,omitempty"`
	CreatedTime          *time.Time `json:"createdTime,omitempty"`
	Quantity             float64    `json:"quantity"`
	Tuition              float64    `json:"tuition"`
}

type TuitionAccountFlowRecordListResult struct {
	List  []TuitionAccountFlowRecordItem `json:"list"`
	Total int                            `json:"total"`
}

File: platform-admin/src/api/finance-center/tuition-account-flow.ts
import axios from 'axios'
import { STORAGE_AUTHORIZE_KEY, useAuthorization } from '~/composables/authorization'
import { usePost } from '~/utils/request'

export interface TuitionAccountFlowRecordItem {
  tutionAccountFlowId: string
  tuitionAccountId: string
  studentId: string
  studentName: string
  studentPhone: string
  studentAvatar?: string
  teachingCourseId?: string
  teachingCourseName?: string
  productId: string
  productName: string
  lessonType?: number
  lessonChargingMode?: number
  sourceType: number
  sourceId: string
  teachingRecordId?: string
  createdTime?: string
  quantity: number
  tuition: number
}

export interface TuitionAccountFlowRecordListResult {
  list?: TuitionAccountFlowRecordItem[]
  total?: number
}

export interface SubTuitionAccountFlowRecordItem {
  id: string
  tuitionAccountId: string
  studentId: string
  studentName: string
  studentPhone: string
  studentAvatar?: string
  teachingCourseId?: string
  teachingCourseName?: string
  productId: string
  productName: string
  lessonType?: number
  lessonChargingMode?: number
  sourceType: number
  sourceId: string
  teachingRecordId?: string
  createdTime?: string
  quantity: number
  tuition: number
  balanceQuantity: number
  balanceTuition: number
  orderNumber?: string
}

export interface SubTuitionAccountFlowRecordListResult {
  list?: SubTuitionAccountFlowRecordItem[]
  total?: number
}

export interface TuitionAccountFlowRecordListQueryParams {
  queryModel?: {
    tuitionAccountId?: string
    productId?: string
    studentId?: string
    orderNumber?: string
    sourceTypes?: number[]
    startTime?: string
    endTime?: string
  }
  pageRequestModel: {
    needTotal?: boolean
    pageSize: number
    pageIndex: number
    skipCount?: number
  }
  sortModel?: {
    orderByCreatedTime?: number
  }
}

export function getTuitionAccountFlowRecordListApi(data: TuitionAccountFlowRecordListQueryParams) {
  return usePost<TuitionAccountFlowRecordListResult>('/api/v1/tuition-account-flows/list', data)
}

export function getSubTuitionAccountFlowRecordListApi(data: TuitionAccountFlowRecordListQueryParams) {
  return usePost<SubTuitionAccountFlowRecordListResult>('/api/v1/tuition-account-flows/sub-list', data)
}

export async function exportTuitionAccountFlowRecordListApi(data: {
  queryModel?: TuitionAccountFlowRecordListQueryParams['queryModel']
  sortModel?: TuitionAccountFlowRecordListQueryParams['sortModel']
}) {
  const token = useAuthorization()
  return axios.post('/api/v1/tuition-account-flows/export', data, {
    responseType: 'blob',
    headers: {
      [STORAGE_AUTHORIZE_KEY]: token.value || '',
      Authorization: token.value ? `Bearer ${token.value}` : '',
      'Accept-Language': 'zh-CN',
    },
  })
}

export async function exportSubTuitionAccountFlowRecordListApi(data: {
  queryModel?: TuitionAccountFlowRecordListQueryParams['queryModel']
  sortModel?: TuitionAccountFlowRecordListQueryParams['sortModel']
}) {
  const token = useAuthorization()
  return axios.post('/api/v1/tuition-account-flows/sub-export', data, {
    responseType: 'blob',
    headers: {
      [STORAGE_AUTHORIZE_KEY]: token.value || '',
      Authorization: token.value ? `Bearer ${token.value}` : '',
      'Accept-Language': 'zh-CN',
    },
  })
}
